packages/infrastructure-lib/src/mongo_settlement_matrix_repo.ts

```diff
@@ -101,15 +104,71 @@ export class MongoSettlementMatrixRepo implements ISettlementMatrixRequestRepo {
 		}
 	}
 
-	async getMatrices(state?:string): Promise<ISettlementMatrix[]>{
+	async getMatrices(
+		matrixId?: string,
+		type?: string,
+		state?: string,
+		model?: string,
+		currencyCodes?: string[],
+		createdAt?: string,
+		pageIndex: number = 0,
+        pageSize: number = MAX_ENTRIES_PER_PAGE,
+	): Promise<MatrixSearchResults> {
 		try {
-			let resp;
-			if(state){
-				resp = await this._collection.find({state: state}).project({_id: 0}).toArray();
-			}else{
-				resp = await this._collection.find().project({_id: 0}).toArray();
+			// Pagination settings
+			pageIndex = Math.max(pageIndex, 0);
+			pageSize = Math.min(pageSize, MAX_ENTRIES_PER_PAGE);
+			const skip = pageIndex * pageSize;
+
+			const filter = [];
+			if (matrixId) filter.push({ id: matrixId });
+			if (type) filter.push({ type: type });
+			if (state) filter.push({ state: state });
+			if (model) filter.push({ settlementModel: model });
+			if (currencyCodes && currencyCodes.length > 0) filter.push({ currencyCodes: { $in: currencyCodes } });
+			if (createdAt) filter.push({ createdAt: { $eq: Number(createdAt) } });
+
+			let match = {};
+			if (filter.length > 0) match = { $and: filter };
+
+			const pipeline = [
```

Comment:
Same as above, use transfers implementation as reference (2 queries, not an aggregate query)